### PR TITLE
[Kotlin Migration] DeckOptions

### PR DESCRIPTION
**Pull request by BrayanDSO**
Definetely not my favorite class. Tested manually on my phone.
There were too many deprecations (even on imports) so I suppressed the whole file.

## Checklist
_Please, go through these checks before submitting the PR._

- [X] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [X] You have a descriptive commit message with a short title (first line, max 50 chars).
- [X] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [X] You have commented your code, particularly in hard-to-understand areas
- [X] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckOptions.kt [hunk 1]
```diff
@@ -17,918 +14,795 @@ package com.ichi2.anki;
...
             } else {
-                value = this.mPref.getString(key, "");
+                mPref.getString(key, "")
             }
             // update summary
             if (!mPref.mSummaries.containsKey(key)) {
-                CharSequence s = pref.getSummary();
-                mPref.mSummaries.put(key, s != null ? pref.getSummary().toString() : null);
+                val s = pref.summary
+                mPref.mSummaries[key] = if (s != null) pref.summary.toString() else null
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckOptions.kt:669]: cleanup: ?.

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckOptions.kt [hunk 2]
```diff
@@ -17,918 +14,795 @@ package com.ichi2.anki;
...
-            Deck child = mCol.getDecks().get(childDid);
-            if (child.isDyn()) {
-                continue;
+    private val subdeckCount: Int
+        get() {
+            var count = 0
+            val did = mDeck.getLong("id")
+            val children = mCol!!.decks.children(did)
+            @KotlinCleanup("invert if")
+            for (childDid in children.values) {
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckOptions.kt:765]: or change to a `.sum {}`

**Review by david-allison — CHANGES_REQUESTED** [review]
If it makes you feel better the 'non-deprecated' preferences library is pretty awful. I've done the main 'preferences' activity, and will do these in time.

Soft approve as both changes are minor and the code seems correct.

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckOptions.kt (hunk 2 above)
Comment (BrayanDSO) [reply AnkiDroid/src/main/java/com/ichi2/anki/DeckOptions.kt:765]: Do you mean `.count {}`? As in:

```kotlin
children.values.count { !mCol!!.decks.get(it).isDyn }
```

Comment (david-allison) [reply AnkiDroid/src/main/java/com/ichi2/anki/DeckOptions.kt:766]: Oh -  if we're only incrementing by 1 then count would be better yes!

**Review by mikehardy — APPROVED** [review]
boom!

**Review by david-allison — APPROVED** [review]

**Pull request merged**